src/pages/dataElements/form/createDataElementSchema.ts

```diff
@@ -0,0 +1,122 @@
+import i18n from '@dhis2/d2-i18n'
+import { z } from 'zod'
+import type { ModelSchemas } from '../../../lib'
+
+const requiredMessage = i18n.t('Required')
+const max50Message = i18n.t('Cannot be longer than {{number}} character', {
+    number: 50,
+})
+
+export const createDataElementSchema = (schemas: ModelSchemas) =>
+    z.object({
+        name: z
+            .string()
+            .min(1, { message: requiredMessage })
+            .max(schemas.dataElement.properties.name.length || 255, {
+                message: max50Message,
+            }),
+        shortName: z
+            .string()
+            .min(1, { message: requiredMessage })
+            .max(schemas.dataElement.properties.shortName.length || 255, {
+                message: max50Message,
+            }),
+        code: z.string().optional(),
+        description: z
+            .string()
+            .max(255, {
+                message: i18n.t(
+                    'The value is to long. You can use up to 255 characters'
+                ),
+            })
+            .optional(),
+        url: z.string().optional(),
+        style: z.object({
+            color: z.string().optional(),
+            icon: z.string().optional(),
+        }),
+        icon: z.string().optional(),
+        fieldMask: z.string().optional(),
+        domainType: z.union([z.literal('AGGREGATE'), z.literal('TRACKER')]),
+        formName: z.string().optional(),
+        valueType: z
```

Comment:
Im pretty sure you can just use the enum in the generated types here. `valueType: z.nativeEnum(DataElement.valueType)`

Reply:
I'm hesitant to change anything in generated files until we can guarantee that the change will persist even when re-generating the files.

Reply:
Currently it's just copied from a local repo https://github.com/Birkbjo/dhis2-open-api-ts , I will add that there.

Reply:
Done